ml/src/main/scala/org/apache/spark/ml/h2o/param/H2OTargetEncoderParams.scala

```diff
@@ -0,0 +1,125 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.apache.spark.ml.h2o.param
+
+import org.apache.spark.h2o.utils.H2OSchemaUtils
+import org.apache.spark.ml.h2o.features._
+import org.apache.spark.ml.PipelineStage
+import org.apache.spark.ml.param.{Param, Params, StringArrayParam}
+import org.apache.spark.sql.types.{DoubleType, StructField, StructType}
+import org.apache.spark.h2o.Frame
+
+trait H2OTargetEncoderParams extends PipelineStage with Params {
```

Comment:
We don't want to introduce new `PipelineStage`. That might break reading and loading of spark pipelines. 

Users should work with native Spark types such as `Pipeline`, `PipelineModel` and `PipelineStage`

Reply:
@jakubhava It would not break anything.... you can inherit from `PipelineStage` and place some shared logic there (e.g. `transformSchema`).  Liskov substitution principle.

Reply:
yes, this will work, thanks. Got confused as previously we had H2OPipeline in the code which was really breaking serialization and deserialization + user was forced to use different kind of pipeline.

As long as the introduced types are not exposed, we're good